plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3select/TestS3SelectRecordCursorProvider.java

```diff
@@ -0,0 +1,158 @@
+        assertTrue(recordCursor.isPresent());
+    }
+
+    @Test
+    public void shouldNotReturnSelectRecordCursorWhenPushdownIsDisabled()
+    {
+        s3SelectPushdownEnabled = false;
+        List<HiveColumnHandle> columnHandleList = new ArrayList<>();
+        TupleDomain<HiveColumnHandle> effectivePredicate = TupleDomain.all();
+        Optional<HiveRecordCursorProvider.ReaderRecordCursorWithProjections> recordCursor =
+                S3_SELECT_RECORD_CURSOR_PROVIDER.createRecordCursor(
+                        CONFIGURATION, SESSION, PATH, START, LENGTH, FILESIZE, SCHEMA, columnHandleList, effectivePredicate, TESTING_TYPE_MANAGER, s3SelectPushdownEnabled);
+        assertTrue(recordCursor.isEmpty());
+    }
+
+    @Test
+    public void shouldNotReturnSelectRecordCursorWhenQueryIsNotFiltering()
+    {
+        s3SelectPushdownEnabled = true;
+        TupleDomain<HiveColumnHandle> effectivePredicate = TupleDomain.all();
+        Optional<HiveRecordCursorProvider.ReaderRecordCursorWithProjections> recordCursor =
+                S3_SELECT_RECORD_CURSOR_PROVIDER.createRecordCursor(
+                        CONFIGURATION, SESSION, PATH, START, LENGTH, FILESIZE, SCHEMA, SCHEMA_COLUMNS, effectivePredicate, TESTING_TYPE_MANAGER, s3SelectPushdownEnabled);
+        assertTrue(recordCursor.isEmpty());
+    }
+
+    @Test
+    public void shouldNotReturnSelectRecordCursorWhenProjectionOrderIsDifferent()
+    {
+        s3SelectPushdownEnabled = true;
+        TupleDomain<HiveColumnHandle> effectivePredicate = TupleDomain.all();
+        final List<HiveColumnHandle> readerColumns = ImmutableList.of(DATE_ARTICLE_COLUMN, QUANTITY_COLUMN, ARTICLE_COLUMN, AUTHOR_COLUMN);
+        Optional<HiveRecordCursorProvider.ReaderRecordCursorWithProjections> recordCursor =
+                S3_SELECT_RECORD_CURSOR_PROVIDER.createRecordCursor(
+                        CONFIGURATION, SESSION, PATH, START, LENGTH, FILESIZE, SCHEMA, readerColumns, effectivePredicate, TESTING_TYPE_MANAGER, s3SelectPushdownEnabled);
+        assertTrue(recordCursor.isEmpty());
+    }
+
+    private static Properties createTestingSchema()
+    {
+        Properties schema = new Properties();
+        String columnNames = buildPropertyFromColumns(SCHEMA_COLUMNS, HiveColumnHandle::getName);
+        String columnTypeNames = buildPropertyFromColumns(SCHEMA_COLUMNS, column -> column.getHiveType().getTypeInfo().getTypeName());
+        schema.setProperty(LIST_COLUMNS, columnNames);
+        schema.setProperty(LIST_COLUMN_TYPES, columnTypeNames);
+        String deserializerClassName = LazySimpleSerDe.class.getName();
+        schema.setProperty(SERIALIZATION_LIB, deserializerClassName);
+        return schema;
+    }
+
+    private static String buildPropertyFromColumns(List<HiveColumnHandle> columns, Function<HiveColumnHandle, String> mapper)
+    {
+        if (columns == null || columns.isEmpty()) {
```

Comment:
can `columns` ever be null?

Reply:
it shouldn't be, i am dropping the null check. But they can be empty.